packages/core/src/utils/ts-helpers.ts

```diff
@@ -0,0 +1,6 @@
+/**
+ * Remove Index Signature
+ */
+export type RemoveIndexSignature<T> = {
+  [K in keyof T as {} extends Record<K, 1> ? never : K]: T[K]
```

Comment:
I do not fully understand how this works (why the `1`?) but that's really neat that it does.

Reply:
Us both!